File: undercover_mobile/lib/src/models/artist_tweet.dart
import 'dart:convert';

List<ArtistTweet> artistTweetFromJson(String str) => List<ArtistTweet>.from(
    json.decode(str).map((x) => ArtistTweet.fromJson(x)));

String artistTweetToJson(List<ArtistTweet> data) =>
    json.encode(List<dynamic>.from(data.map((x) => x.toJson())));

class ArtistTweet {
  ArtistTweet({
    required this.createdAt,
    required this.idStr,
    required this.text,
    required this.truncated,
    required this.retweetCount,
    required this.favoriteCount,
    required this.favorited,
    required this.retweeted,
    required this.lang,
  });

  factory ArtistTweet.fromJson(Map<String, dynamic> json) => ArtistTweet(
        createdAt: json['created_at'],
        idStr: json['id_str'],
        text: json['text'],
        truncated: json['truncated'],
        retweetCount: json['retweet_count'],
        favoriteCount: json['favorite_count'],
        favorited: json['favorited'],
        retweeted: json['retweeted'],
        lang: json['lang'],
      );

  final String createdAt;
  final String idStr;
  final String text;
  final bool truncated;
  final int retweetCount;
  final int favoriteCount;
  final bool favorited;
  final bool retweeted;
  final String lang;

  Map<String, dynamic> toJson() => {
        'created_at': createdAt,
        'id_str': idStr,
        'text': text,
        'truncated': truncated,
        'retweet_count': retweetCount,
        'favorite_count': favoriteCount,
        'favorited': favorited,
        'retweeted': retweeted,
        'lang': lang,
      };
}

File: WebAPI/Undercover/Undercover.API/Models/ArtistTweet.cs
using Newtonsoft.Json;
using System.Collections.Generic;

namespace Undercover.API.Models
{
    public partial class ArtistTweet
    {
        [JsonProperty("created_at")]
        public string CreatedAt { get; set; }

        [JsonProperty("id_str")]
        public string IdStr { get; set; }

        [JsonProperty("text")]
        public string Text { get; set; }

        [JsonProperty("truncated")]
        public bool Truncated { get; set; }

        [JsonProperty("retweet_count")]
        public long RetweetCount { get; set; }

        [JsonProperty("favorite_count")]
        public long FavoriteCount { get; set; }

        [JsonProperty("favorited")]
        public bool Favorited { get; set; }

        [JsonProperty("retweeted")]
        public bool Retweeted { get; set; }

        [JsonProperty("lang")]
        public string Lang { get; set; }

    }

    public partial class ArtistTweet
    {
        public static List<ArtistTweet> FromJson(string json) => JsonConvert.DeserializeObject<List<ArtistTweet>>(json);
    }

    public static class Serialize
    {
        public static string ToJson(this List<ArtistTweet> self) => JsonConvert.SerializeObject(self);
    }
}
